tests/ludwig/config_validation/test_checks.py

```diff
@@ -11,6 +11,7 @@
 from typing import Any, Dict, Optional
 
 import pytest
+from marshmallow import ValidationError
```

Comment:
You should be able to just use `ConfigValidationError`

Reply:
It fails with the marshmallow ValidationError

Reply:
That's strange. The code should be wrapping it in a `ConfigValidationError`. CC @justinxzhao

Reply:
Actually glancing through the code we manually raise a `ValidationError` in quite a few locations. (Worth a separate refactoring.)

Reply:
ConfigValidationError is only raised at the ModelConfig level.

Reply:
Agreed it might be good to clarify some of our semantics about ConfigValidationError vs. ValidationError from marshmallow, perhaps moving all Ludwig-specific validation code to `ConfigValidationError`.

@abidwael It would be good to understand how the test gets through raises a marshmallow `ValidationError`. Perhaps it's because we don't `except` it in [this block](https://github.com/ludwig-ai/ludwig/blob/f26222046f8415b465944823ad7d0605ac1f40a2/ludwig/config_validation/validation.py#L62-L66).

Otherwise, LGTM.

Reply:
Ah @justinxzhao thanks for the link, I totally missed that <imblind.gif>